ktor-shared/ktor-websockets/jvm/test/WebSocketReaderTest.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.websocket
+
+import io.ktor.utils.io.*
+import io.ktor.utils.io.core.*
+import io.ktor.websocket.*
+import kotlinx.coroutines.*
+import kotlin.test.*
+
+class WebSocketReaderTest {
+    @Test
+    fun testFailFragmentedControl() {
```

Comment:
can you also add a test for non final control frame, please?

Reply:
Unfortunately I'm not sure that's actually possible due to `Frame`'s design. I can't create raw frame or subclass it due to `sealed` and the existing control frames don't allow me to set `fin`. Any advice how to make that work?

Reply:
You are right. It would be possible to create a serialized representation of the frame and send it to parser. But not worth it.

Reply:
You are right. It would be possible to create a serialized representation of the frame and send it to parser. But not worth it.